be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -105,6 +105,168 @@ private:
     }
 };
 
+template <PrimitiveType PT>
+class ArrayIntersect {
+public:
+    using CppType = RunTimeCppType<PT>;
+
+    class CppTypeWithOverlapTimes {
+    public:
+        CppTypeWithOverlapTimes(const CppType& item, size_t n = 0) : value(item), overlap_times(n) {}
+
+        CppType value;
+        mutable size_t overlap_times;
+    };
+
+    template <size_t type>
```

Comment:
type is a trivial name?

Reply:
what type?

Reply:
> type is a trivial name?

Yes, I will rename this

Reply:
> what type?

different type is just for different column type, It's actually label name. I will refact to indicate this.